### ISSUE TITLE
access token payload property is omitted when value is 'null'

### ISSUE THREAD

**Issue by n1ru4l** [supertokens/supertokens-core#499]
## 🐛 Bug Report

I noticed that when I decode the access token payload a `null` value property is actually omitted. I expected it to keep the value `null` within the payload.

At the same time within the `sessionData` the null value exists in the payload (within the  database).

```ts
const SuperTokenAccessTokenModel = zod.object({
  version: zod.literal('1'),
  superTokensUserId: zod.string(),
  /**
   * Supertokens for some reason omits externalUserId from the access token payload if it is null.
   */
  externalUserId: zod.optional(zod.union([zod.string(), zod.null()])),
  email: zod.string(),
});
```

## Useful informations

```ts
SessionNode.init({
  override: {
    functions: originalImplementation => {
      return {
        ...originalImplementation,
        createNewSession: async function (input) {
          const user = await ThirdPartyEmailPasswordNode.getUserById(input.userId);

          if (!user) {
            throw new Error(
              `SuperTokens: Creating a new session failed. Could not find user with id ${input.userId}.`
            );
          }

          const externalUserId = user.thirdParty ? `${user.thirdParty.id}|${user.thirdParty.userId}` : null;

          input.accessTokenPayload = {
            version: '1',
            superTokensUserId: input.userId,
            externalUserId, // This is potentially null
            email: user.email,
          };

          input.sessionData = {
            version: '1',
            superTokensUserId: input.userId,
            externalUserId,
            email: user.email,
          };

          return originalImplementation.createNewSession(input);
        },
      };
    },
  },
})
```

**Comment by porcellus** [comment]
Hi, thanks for reporting this. This could be a bug in our current version.
However, we are planning to move to "merging" access token payloads instead of replacing the entire object in the near future.
Does your use case require keeping a null property in the root of the access token payload, or would you be OK with putting your user info into an object inside the payload?

**Comment by n1ru4l** [comment]
> Does your use case require keeping a null property in the root of the access token payload, or would you be OK with putting your user info into an object inside the payload?

I can also put it out of the root, but coming mostly form a JavaScript background I did not expected a `null` value being dropped.

E.g. if you call `JSON.stringify` you get the following results:

`JSON.stringify({ foo: null })` -> `'{"foo":null}'`
`JSON.stringify({ foo : undefined })` -> `'{}'`

> However, we are planning to move to "merging" access token payloads instead of replacing the entire object in the near future.

Not sure what this actually means and how this solves this issue. 😅 If it confirms JSON serialization of JavaScript object I agree with it!

**Comment by rishabhpoddar** [comment]
That being said, you can continue to use updateAccessTokenPayload (though eventually that function will be removed) and we are also planning to add an argument to mergeIntoAccessTokenPayload which would allow you to add top level nulls. So it should all just work fine.

**Comment by rishabhpoddar** [comment]
This should be fixed in version >= 3.16.2.

If you are using docker, then the tag is 3.16.
